Lib/test/support/__init__.py

```diff
@@ -1982,3 +1983,13 @@ def skip_if_broken_multiprocessing_synchronize():
             synchronize.Lock(ctx=None)
         except OSError as exc:
             raise unittest.SkipTest(f"broken multiprocessing SemLock: {exc!r}")
+
+
+def check_disallow_instantiation(testcase, tp, *args, **kwds):
+    """
+    Helper for testing types with the Py_TPFLAGS_DISALLOW_INSTANTIATION flag.
+
+    See bpo-43916.
+    """
+    msg = f"cannot create '{tp.__module__}\.{tp.__name__}' instances"
```

Comment:
You should use f"xxx\\.xxx" or fr"xxx\.xxx". Or maybe use re.escape().

Reply:
Since these tests are decorated with `cpython_only`, it should be ok to match the exact string, as produced by Objects/typeobject.c?

Reply:
What about setting the match pattern depending on what `test.support.check_impl_detail()` says?